packages/rollup-plugin-html/src/output/injectServiceWorkerRegistration.ts

```diff
@@ -0,0 +1,47 @@
+import path from 'path';
+import { Document } from 'parse5';
+import {
+  findElement,
+  getTagName,
+  appendChild,
+  createScript,
+  setAttribute,
+} from '@web/parse5-utils';
+
+export interface injectServiceWorkerRegistrationArgs {
+  document: Document;
+  swDest: string;
+  outputDir: string;
+  htmlFileName: string;
+}
+
+export function injectServiceWorkerRegistration(args: injectServiceWorkerRegistrationArgs) {
+  const { document, swDest, outputDir, htmlFileName } = args;
+  const body = findElement(document, e => getTagName(e) === 'body');
+  if (!body) {
+    throw new Error('Missing body in HTML document.');
+  }
+
+  let swPath = swDest.replace(`${outputDir}${path.sep}`, '');
+  swPath = path.relative(path.dirname(htmlFileName), swPath);
```

Comment:
This doesn't take into account windows file paths. We also need to take into account the `publicPath` option here. 

The logic for generating the right path is already in the `injectedUpdatedAssetPaths` function. So perhaps we can pull that to a separate util?

Reply:
hmmm windows test now succeed?

I tried to apply the same logic but there is no relativeOutputPath (e.g. that is what we want to get... but that logic needs it as an input? 🤔)

Reply:
What the user puts in the option is a file system path, right? so on windows it could be something like `C:\\foo\bar` which needs to be transformed to a browser path

Reply:
see below